src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+                        }
+                    } else {
+                        this->textId = 0x5EC;
+                        gSaveContext.weekEventReg[17] |= 0x10;
+                    }
+                    break;
+                case PLAYER_FORM_HUMAN:
+                    if (Player_GetMask(globalCtx) == PLAYER_MASK_KAFEIS_MASK) {
+                        this->textId = 0x236A;
+                    } else if (gSaveContext.weekEventReg[74] & 0x10) {
+                        this->textId = 0x5F6;
+                    } else {
+                        this->textId = 0x5F5;
+                        gSaveContext.weekEventReg[74] |= 0x10;
+                    }
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    if (gSaveContext.weekEventReg[75] & 8) {
+                        this->textId = 0x5E4;
+                    } else {
+                        this->textId = 0x5DC;
+                        gSaveContext.weekEventReg[75] |= 8;
+                    }
+                    break;
+            }
+            break;
+        case ENBJI01_PARAMS_LOOKED_THROUGH_TELESCOPE:
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if (gSaveContext.weekEventReg[74] & 0x80) {
+                        this->textId = 0x5F2;
+                    } else {
+                        this->textId = 0x5F1;
+                    }
+                    func_800B8500(&this->actor, globalCtx, this->actor.xzDistToPlayer, this->actor.yDistToPlayer, 0);
+                    break;
+                case PLAYER_FORM_HUMAN:
+                    this->textId = 0x5F7;
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    switch (CURRENT_DAY) {
+                        case 1:
+                            this->textId = 0x5E9;
+                            break;
+                        case 2:
+                            this->textId = 0x5EA;
+                            break;
+                        case 3:
+                            tempDay = gSaveContext.day;
+                            tempTimeBeforeMoonCrash =
+                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
+                            if (tempTimeBeforeMoonCrash < 2730.6667f) { /* 1 hr */
+                                this->textId = 0x5E8;
+                            } else {
+                                this->textId = 0x5EB;
+                            }
+                            break;
+                    }
+            }
+            break;
+    }
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 2, &this->animationIndex);
+    this->actionFunc = func_809CD328;
+}
+
+void func_809CD328(EnBji01* this, GlobalContext* globalCtx) {
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
```

Comment:
Just as a note, function `func_80152498` is some kind of "GetDialogueState" (or whatever), in the sense the number it returns tells you the current state of the conversation (player answered a question, a message block ended, the whole conversation ended, etc).

We usually name this kind of function `_DialogueHandler`.

EnMa4 (https://github.com/zeldaret/mm/blob/9637e19303bdd2c99b1a03390cc1bbea786a8150/src/overlays/actors/ovl_En_Ma4/z_en_ma4.c#L649) has an example of this and a few notes on the values it uses. I though you may be interested on knowing this.

Reply:
Awesome! I'll rename it when i get home.

Reply:
Just ran rename_sym.sh and pushed it :)